File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map <int,int> mp;
        vector<int> res;
        for(int i =0;i<n;i++)
        {
            
            {
                if(mp.find(target-nums[i])!=mp.end())
                {
                    return{i,mp[target-nums[i]]} ;
                }
                else mp[nums[i]]=i;
            }
        }
        return res;
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        ArrayList<Integer> l = new ArrayList<>();
        for(int i =0;i<nums.length;i++)
        {
            if(mp.containsKey(target-nums[i]))
            {
                int x = mp.get(target-nums[i]);
                return new int []{x, i};
            }
            else mp.put(nums[i],i);
        }
        return new int[]{};
    }
}